src/Core/Grid/Definition/Factory/ProductGridDefinitionFactory.php

```diff
@@ -337,4 +400,50 @@ protected function getGridActions()
                     ->setIcon('storage')
             );
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getBulkActions()
+    {
+        return (new BulkActionCollection())
+            ->add(
+                (new AjaxBulkAction('enable_selection_ajax'))
+                    ->setName($this->trans('Activate selection', [], 'Admin.Actions'))
+                    ->setOptions([
+                        'ajax_route' => 'admin_products_v2_active_ajax',
+                        'modal_title' => $this->trans('Activating products', [], 'Admin.Actions'),
+                        'modal_progress_title' => $this->trans('Activating...', [], 'Admin.Actions'),
+                        'modal_confirm_button_label' =>  $this->trans('Confirm', [], 'Admin.Actions'),
+                        'modal_close_button_label' => $this->trans('Close', [], 'Admin.Actions'),
+                        'modal_confirm_message' => $this->trans('Product activation is in progress, please don\'t close the window', [], 'Admin.Actions')
+                    ])
+            )
+            ->add(
+                (new SubmitBulkAction('enable_selection'))
+                    ->setName($this->trans('Activate selection', [], 'Admin.Actions'))
+                    ->setOptions([
+                        'submit_route' => 'admin_products_v2_bulk_enable',
+                    ])
+            )
+            ->add(
+                (new SubmitBulkAction('disable_selection'))
+                    ->setName($this->trans('Deactivate selection', [], 'Admin.Actions'))
+                    ->setOptions([
+                        'submit_route' => 'admin_products_v2_bulk_disable',
+                    ])
+            )
+            ->add(
+                (new SubmitBulkAction('duplicate_selection'))
```

Comment:
@jolelievre
![Výstřižek](https://user-images.githubusercontent.com/6097524/150799666-00226ef1-fde3-4da4-9986-31b15d1a2831.JPG)
 
 I think we should implement icon and separator support, wdyt?

Reply:
Yep, totally But I suggest we do this in a separate PR